dev.md

```diff
@@ -0,0 +1,116 @@
+# Contributing to the Yunohost core
+
+You wish to implement a new feature in the Yunohost core, but don't know how to
+proceed ? This guide takes you through the various steps of the development and
+contribution process. 
+
+## Setting up a development environment
+
+- **Use [ynh-dev](https://github.com/YunoHost/ynh-dev)** (see the README) to
+  setup a development environment - locally in a virtual machine, or on a VPS.
+  This will setup a working Yunohost instance, using git repositories and
+  symlinks. That way, you will be able to edit files, test your changes, commit
+  stuff and push/pull directly from your development environment.
+
+- Implement and test your feature. Depending on what you want to develop, you
+  will want to :
+   - **Python/CLI core** : work in `/vagrant/yunohost/`, i.e. the Python/CLI core
+   - **Web administration interface** : work in `/vagrant/yunohost-admin/`
+   - You can also work on the other projects (SSOwat, moulinette) in similar ways
+
+## Developping on the Yunohost Python/CLI core
+
+- Work in `/vagrant/yunohost/`
+
+- The actionsmap file (`data/actionsmap/yunohost.yml`) defines the various
+  categories, actions and arguments of the yunohost CLI. Define how you want
+  users to use your feature, and add/edit the corresponding categories, actions
+  and arguments. For example in `yunohost domain add some.domain.tld`, the
+  category is `domain`, the action is `add`, and `some.domain.tld` is an
+  argument.
+
+- Moulinette will automatically map commands in the actionsmap to python
+  functions (and their arguments) located in `src/yunohost/`. For example, typing
+  `yunohost domain add some.domain.tld` will call the function
+  `domain_add(domainName)` in `domain.py`, with the argument `domainName` equal
+  to `"some.domain.tld"`.
+
+#### Helpers / coding style
+
+- To handle exceptions, you should raise some `MoulinetteError()`
+
+- To help with internationalizing the messages, use `m18n.n('some-message-id')`
+  and put your string in `locales/en.json`. You can also put arguments and use
+  them in the string with `{{some-argument:s}}`. Don't edit other locales files,
+  only the en.json !
+
+- Yunohost tries to follow the [pep8](http://pep8.org/) coding style. Tools
+  exist to automatically check conformity.
+
+- Name of "private" functions should start with a `_`
+
+#### Don't forget
+
+- (Might not be necessary anymore) Each time you edit the actionsmap, you should
+  force the refresh of the cache with `rm
+  /var/cache/moulinette/actionsmap/yunohost.pkl`
+
+## Adding stuff to the Yunohost web administration interface
+
+- Work in `/vagrant/yunohost-admin/src/`
+
+- Make sure the web admin interface of your development environment works, by
+  opening `https://domain.tld/yunohost/admin`. If not, run `npm install` and
+  `npm run build` in `yunohost-admin/src/`. [Not sure about this, to be checked]
+
```

Comment:
I tried https://github.com/YunoHost/yunohost-admin/pull/136
And get this issue:
```bash
y-uniq@1.0.3, replace-ext@0.0.1, has-gulplog@0.1.0, fancy-log@1.2.0, gulplog@1.0.0, lodash.template@3.6.2, vinyl@0.5.3, through2@2.0.1, multipipe@0.1.2, dateformat@1.0.12)
└── vinyl-fs@0.3.14 (strip-bom@1.0.0, graceful-fs@3.0.11, defaults@1.0.3, vinyl@0.4.6, mkdirp@0.5.1, through2@0.6.5, glob-stream@3.1.18, glob-watcher@0.0.6)
No passwd entry for user 'vagrant'
No passwd entry for user 'vagrant'
Using Git repository for yunohost-admin
No passwd entry for user 'vagrant'
```

With the admin broken: